src/dali_backend.cc

```diff
@@ -436,14 +444,30 @@ TRITONBACKEND_ModelInstanceExecute(
           "Unknown DALI Backend error");
     }
 
+    exec_end_ns = detail::capture_time();
+
+    RETURN_IF_ERROR(TRITONBACKEND_ModelInstanceReportStatistics(
```

Comment:
Can't return from this function until all responses have been sent. See identity example. This function can return error only if no responses are sent. But as soon as any response is sent (eigher success or error) then all responses must be sent and the function must return success.

So, the one RETURN_IF_ERROR in the loop above and the three RETURN_IF_ERROR here must be changed to LOG_IF_ERROR.

Reply:
Done. I've left `RETURN_IF_ERROR` only for `ModelInstanceState` call.